tests/main/snap-service/task.yaml

```diff
@@ -25,3 +25,29 @@ execute: |
     else
         systemctl status snap.test-snapd-service.test-snapd-service-refuses-to-stop|MATCH "Result: timeout"
     fi
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    
+    echo "A snap that has install-mode: disable will not start by default"
+    snap services | MATCH 'test-snapd-service.test-snapd-service-install-mode-disable\s+disabled\s+inactive'
+
+    echo "And after a refresh the service stays disabled"
+    install_local test-snapd-service
+    snap services | MATCH 'test-snapd-service.test-snapd-service-install-mode-disable\s+disabled\s+inactive'
+
+    echo "But install-mode: disable services can be enabled"
+    snap start --enable test-snapd-service.test-snapd-service-install-mode-disable
+    snap services | MATCH 'test-snapd-service.test-snapd-service-install-mode-disable\s+enabled\s+active'
+
+    echo "And after a refresh the service stays enabled"
+    install_local test-snapd-service
+    snap services | MATCH 'test-snapd-service.test-snapd-service-install-mode-disable\s+enabled\s+active'
+    # XXX: add test with a refresh vom v1 to v2
+    # v1 has one services (srv1).
+    # v2 has two services (srv1,srv2):
+    #   srv1 is now disabled
+    #   srv2 is new and disabled
+    # The behavior should be that:
+    #   srv1 stays enabled
+    #   srv2 is disabled 
```

Comment:
We will also need a test where service gets enabled via hook.

Reply:
Thanks, I will look into this